src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0102Tests.cs

```diff
@@ -0,0 +1,203 @@
+    using System.Threading.Tasks;
+    public class MyClass
+    {
+        public static void Foo()
+        {
+            new ValueTask()./*MM*/GetAwaiter().GetResult();
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source, version);
+
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0102", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostic.Location);
+        }
+
+        [Fact]
+        public async Task AZC0102WarningOnAwaitable()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static void Foo()
+        {
+            new CustomAwaitable()./*MM*/GetAwaiter().GetResult();
+        }
+    }
+
+    public class CustomAwaitable
+    {
+        public CustomAwaiter GetAwaiter() => new CustomAwaiter();
+    }
+
+    public class CustomAwaiter : ICriticalNotifyCompletion
+    {
+        internal bool IsCompleted => true;
+        protected internal void GetResult() {}
+        public void OnCompleted(Action continuation) {}
+        public void UnsafeOnCompleted(Action continuation) {}
+    }  
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0102", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostic.Location);
+        }
+
+        [Fact]
+        public async Task AZC0102WarningOnExtension()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static void Foo()
+        {
+            new TestStruct()./*MM*/GetAwaiter().GetResult();
+        }
+    }
+
+    public struct TestStruct { }
+
+    public static class Extensions
+    {
+        public static TaskAwaiter GetAwaiter(this TestStruct s) => default;
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0102", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostic.Location);
+        }
+
+        [Fact]
+        public async Task AZC0102NoWarningOnNonAwaiter()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+    
+    public class MyClass
+    {
+        public static void Foo()
+        {
+            new CustomAwaitable()./*MM*/GetAwaiter().GetResult();
```

Comment:
Why don't we wan't a warning here?

Reply:
Cause that isn't an awaiter. See `CustomAwaitable` below.